src/coreclr/jit/compiler.h

```diff
@@ -9057,6 +9089,41 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         return opts.compSupportsISA.HasInstructionSet(isa);
     }
 
+#ifdef DEBUG
+    //------------------------------------------------------------------------
+    // IsBaselineVector512IsaSupportedDebugOnly - Does the target have isa support required for Vector512.
+    //
+    // Returns:
+    //    `true` if AVX512F, AVX512BW and AVX512DQ are supported.
+    //
+    bool IsBaselineVector512IsaSupportedDebugOnly() const
+    {
+#ifdef TARGET_AMD64
+        return (compIsaSupportedDebugOnly(InstructionSet_AVX512F) &&
+                compIsaSupportedDebugOnly(InstructionSet_AVX512BW) &&
+                compIsaSupportedDebugOnly(InstructionSet_AVX512DQ));
```

Comment:
I think we can actually simplify this to be `compIsaSupportedDebugOnly(InstructionSet_Vector512)`

This should work since `compiler.cpp` only does `instructionSetFlags.AddInstructionSet(InstructionSet_Vector512);` if the three ISAs we're checking are actually supported.

Reply:
Done